src/daft-scan/src/glob.rs

```diff
@@ -187,15 +187,16 @@ impl GlobScanOperator {
         let (partitioning_keys, generated_fields) = if partition_fields.is_empty() {
             (vec![], Schema::empty())
         } else {
-            let generated_fields = Schema::new(partition_fields)?;
+            let generated_fields = Schema::new(partition_fields);
             let generated_fields = match user_provided_schema.clone() {
                 Some(hint) => generated_fields.apply_hints(&hint)?,
                 None => generated_fields,
             };
             // Extract partitioning keys only after the user's schema hints have been applied.
-            let partitioning_keys = (&generated_fields.fields)
-                .into_iter()
-                .map(|(_, field)| PartitionField::new(field.clone(), None, None))
+            let partitioning_keys = generated_fields
+                .fields()
+                .iter()
+                .map(|field| PartitionField::new(field.clone(), None, None))
```

Comment:
unrelated to this particular section, but I'm starting to realize I _think_ making schema an iterator of fields would reduce repetition overall. No need to address in this PR, but something to consider in subsequent.